src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UnitTest/UnitTestsControl.cs

```diff
@@ -202,6 +202,11 @@ private async Task RunTests(CancellationToken cts)
 								{
 									throw new TimeoutException($"Test execution timed out after {DefaultUnitTestTimeout}");
 								}
+
+								if (resultingTask.Exception != null)
+								{
+									throw resultingTask.Exception;
```

Comment:
```suggestion
									ExceptionDispatchInfo.Capture(resultingTask.Exception.InnerException).Throw();
```

Reply:
The stack trace seems reasonably intact? Here is a sample:

![image](https://user-images.githubusercontent.com/8270914/70814738-1dc76f80-1d9a-11ea-9bce-bbe421c6a82c.png)

In the `catch` block it looks like we already fish out the inner exception to report it.

Reply:
Hummmm weird, the stacktrace of an exception is set in the `throw` ... probably that's because you are raising the `AggregateException` which is then unwrap for the test result.